adapters/adyoulike/adyoulike.go

```diff
@@ -0,0 +1,159 @@
+			errs = append(errs, newBadInputError(err.Error()))
+		}
+	}
+
+	if len(openRTBRequest.Imp) == 0 {
+		errs = append(errs, newBadInputError("No impression in the bid request"))
+	}
+
+	reqCopy := *openRTBRequest
+	reqCopy.Imp = []openrtb.Imp{}
+	for ind, imp := range openRTBRequest.Imp {
+		reqCopy.Imp = append(reqCopy.Imp, imp)
+
+		tagID, err = jsonparser.GetString(reqCopy.Imp[ind].Ext, "bidder", "placement")
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		reqCopy.Imp[ind].TagID = tagID
+	}
+
+	openRTBRequestJSON, err := json.Marshal(reqCopy)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	requestToBidder := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint,
+		Body:    openRTBRequestJSON,
+		Headers: headers,
+	}
+	requestsToBidder = append(requestsToBidder, requestToBidder)
+
+	return requestsToBidder, errs
+}
+
+const unexpectedStatusCodeFormat = "" +
+	"Unexpected status code: %d. Run with request.debug = 1 for more info"
+
+func (adapter *AdYouLikeAdapter) MakeBids(
+	openRTBRequest *openrtb.BidRequest,
+	requestToBidder *adapters.RequestData,
+	bidderRawResponse *adapters.ResponseData,
+) (
+	bidderResponse *adapters.BidderResponse,
+	errs []error,
+) {
+	switch bidderRawResponse.StatusCode {
+	case http.StatusOK:
+		break
+	case http.StatusNoContent:
+		return nil, nil
+	case http.StatusBadRequest:
+		err := &errortypes.BadInput{
+			Message: fmt.Sprintf(unexpectedStatusCodeFormat, bidderRawResponse.StatusCode),
+		}
+		return nil, []error{err}
+	default:
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf(unexpectedStatusCodeFormat, bidderRawResponse.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	var openRTBBidderResponse openrtb.BidResponse
+	if err := json.Unmarshal(bidderRawResponse.Body, &openRTBBidderResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(openRTBRequest.Imp))
+	bidResponse.Currency = openRTBBidderResponse.Cur
+	for _, seatBid := range openRTBBidderResponse.SeatBid {
+		for idx := range seatBid.Bid {
+			b := &adapters.TypedBid{
+				Bid:     &seatBid.Bid[idx],
+				BidType: getMediaTypeForImp(seatBid.Bid[idx].ImpID, openRTBRequest.Imp),
+			}
+			bidResponse.Bids = append(bidResponse.Bids, b)
+		}
+	}
+	return bidResponse, nil
+}
+
+// getMediaTypeForBid determines which type of bid.
+func getMediaTypeForImp(impID string, imps []openrtb.Imp) openrtb_ext.BidType {
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range imps {
+		if imp.ID == impID {
+			if imp.Banner == nil && imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeVideo
+			} else if imp.Banner == nil && imp.Native != nil {
+				mediaType = openrtb_ext.BidTypeNative
+			}
```

Comment:
Does this match your bidding servers behavior? Or better yet, does your bidding server include the bid type in its response?

Reply:
No I loop into response's Imps looking out for bid ID to match bid's mediatype.
ImpId is from seatBid.Bid[idx].ImpID
imps array is from openRTBRequest.Imp